game/scripts/vscripts/abilities/electrician/electrician_static_grip.lua

```diff
@@ -226,10 +215,38 @@ if IsServer() then
 --------------------------------------------------------------------------------
 
   function modifier_electrician_static_grip:OnIntervalThink()
+    -- parent = enemy
     local parent = self:GetParent()
+    -- caster = chatterjee
     local caster = self:GetCaster()
     local spell = self:GetAbility()
 
+    local attackSpeedPercent = spell:GetSpecialValueFor("attack_speed_pct") / 100
+
+    if attackSpeedPercent > 0 then
+      -- seconeds per attack, so larger = slower
+      -- percent is out of 100, so at 100 it should * 1, and at 50% it should be *2
+      -- if we turn the percent into a 0-1 then we can use it as a divisor
+      -- seconds / percent, so 1 second per attack at 90% becomes 1.111
+      -- seems right?
+      local secondsPerAttack = caster:GetSecondsPerAttack(false) / attackSpeedPercent
+      self.ellapsedTime = self.ellapsedTime + self.damageInterval
+
+      if self.ellapsedTime > secondsPerAttack then
```

Comment:
every damage interval i check if CJ *should have already attacked*, and if so it fires off the attack and reduces the attack interval from the total elapsed time so that the "remainder" is preserved